gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,52 +94,175 @@ public synchronized ILiferayProject provide( Object adaptable )
     @Override
     public IStatus createNewProject( NewLiferayModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
     {
-        IStatus retval = null;
+        IStatus retval = Status.OK_STATUS;
 
         final String projectName = op.getProjectName().content();
 
         IPath location = PathBridge.create( op.getLocation().content() );
 
+        String className = op.getComponentName().content();
+
+        final String serviceName = op.getServiceName().content();
+
+        final String packageName = op.getPackageName().content();
+
+        ElementList<PropertyKey> propertyKeys = op.getPropertyKeys();
+
+        final List<String> properties = new ArrayList<String>();
+
+        for( PropertyKey propertyKey : propertyKeys )
+        {
+            properties.add( propertyKey.getKeyName() + "=" + propertyKey.getKeyValue() );
+        }
+
         final String projectTemplateName = op.getProjectTemplateName().content();
 
         StringBuilder sb = new StringBuilder();
         sb.append( "create " );
-        sb.append( "-d \"" + location.toFile().getAbsolutePath() +  "\" " );
+        sb.append( "-d \"" + location.toFile().getAbsolutePath() + "\" " );
         sb.append( "-t " + projectTemplateName + " " );
-        sb.append( "\"" + projectName + "\"");
 
-        try
+        if( className != null )
```

Comment:
Separate this code block into two method, one of them that takes the "op" object and generates the BladeCLI command, and then another method that takes the created project folder and imports it into eclipse.